winui/DataGrid/Export-To-Excel.md

```diff
@@ -0,0 +1,653 @@
+MemoryStream stream = new MemoryStream();
+workBook.SaveAs(stream);
+Save(stream, "Sample");
+{% endhighlight %}
+{% endtabs %}
+
+## Styling cells based on CellType in Excel
+
+You can customize the cell styles based on `CellType` by using `GridExportHandler`.
+
+{% tabs %}
+{% highlight c# %}
+var options = new DataGridExcelExportOptions();
+options.GridExportHandler = GridExportHandler;
+var excelEngine = dataGrid.ExportToExcel(dataGrid.View, options);
+var workBook = excelEngine.Excel.Workbooks[0];
+MemoryStream stream = new MemoryStream();
+workBook.SaveAs(stream);
+Save(stream, "Sample");
+
+private static void GridExportHandler(object sender, DataGridExcelExportStartOptions e)
+{
+    if (e.CellType == ExportCellType.HeaderCell)
+    {
+        e.Style.Color = Color.FromArgb(100, 228, 234);
+        e.Style.Font.Color = ExcelKnownColors.White;
+        e.Handled = true;
+    }
+
+    else if (e.CellType == ExportCellType.RecordCell)
+    {
+        e.Style.Color = Color.FromArgb(240, 224, 144);
+        e.Handled = true;
+    }
+
+    else if (e.CellType == ExportCellType.GroupCaptionCell)
+    {
+        e.Style.Color = Color.FromArgb(252, 159, 161);
+        e.Handled = true;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+<img src="Export-to-Excel_images/winui-datagrid-cell-style-customization.png" alt="WinUI DataGrid displays Customized Cell Style based on CellType in Exported Excel" width="100%" Height="Auto"/>
+
+## Cell customization in Excel while exporting
+
+You can customize the cells by setting `CellsExportHandler` in `DataGridExcelExportOptions`.
+
+### Customize cell value while exporting
+
+You can customize the call values while exporting to excel by using `CellsExportHandler` in `DataGridExcelExportOptions`.
+
+{% tabs %}
+{% highlight c# %}
+var options = new DataGridExcelExportOptions();
+options.CellsExportHandler = CellsExportHandler;
+var excelEngine = dataGrid.ExportToExcel(dataGrid.View, options);
+var workBook = excelEngine.Excel.Workbooks[0];
+MemoryStream stream = new MemoryStream();
+workBook.SaveAs(stream);
+Save(stream, "Sample");
+
+private static void CellsExportHandler(object sender, DataGridCellExcelExportOptions e)
+{
+    // Based on the column mapping name and the cell type, we can change the cell 
+    //values while exporting to excel.
+    if (e.CellType == ExportCellType.RecordCell && e.ColumnName == "IsShipped")
+    {
+
+        //if the cell value is True, "Y" will be displayed else "N" will be displayed.
+
+        if (e.CellValue.Equals(true))
+            e.Range.Cells[0].Value = "Y";
+
+        else
+            e.Range.Cells[0].Value = "N";
+        e.Handled = true;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+<img src="Export-to-Excel_images/winui-datagrid-cell-value-customization.png" alt="Customizing Cell Values while Exporting to Excel in WinUI DataGrid" width="100%" Height="Auto"/>
+
+Here, cell values are changed for `Is Shipped` column based on custom condition.
```

Comment:
Here, the cell values are changed for the `Is Shipped` column based on custom condition.

Reply:
Content modified